analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/TupleBinary.cs

```diff
@@ -29,33 +29,25 @@ internal sealed class TupleBinary : BinaryBase<ITupleBinaryOperationWrapper>
 
     protected override SymbolicConstraint BoolConstraintFromOperation(ProgramState state, ITupleBinaryOperationWrapper operation, bool isLoopCondition, int visitCount)
     {
-        var isUnknown = false;
-        foreach (var pair in TupleElements(operation.LeftOperand, operation.RightOperand))
+        var boolConstraints = TupleElements(operation.LeftOperand, operation.RightOperand)
+            .Select(x => LearnBoolConstraint(BinaryOperatorKind.Equals, state[x.Left], state[x.Right], isLoopCondition, visitCount))
+            .ToArray();
+        if (boolConstraints.Contains(BoolConstraint.False))
```

Comment:
Functional instead of imperative! Very nice.
But remove the `.ToArray()` and replace `Contains` with `Any` to keep it lazily executed.

Reply:
I check the collection twice:
- whether any element is false
- then to check if all elements are true
So I'd be evaluating twice if I used lazy evaluation.